### PR TITLE
Camel case should be permitted in state names and event names

### PR DESCRIPTION
**Pull request by steveluscher**
Given a state name with capital letters in its midst, such as "pendingActivation", a properly spelled `isPendingActivation` method should be generated. Given an event name with an uppercase letter in its midst, such as "makeAdmin", a properly spelled `canMakeAdmin` method should be generated.

### CODE REVIEW

File: StateMachineTests/StateMachineSpec.m [hunk 1]
```diff
@@ -164,22 +164,23 @@ - (void) stopBilling {
...
     });
     
     describe(@"checking if it's in a state", ^{
-        describe(@"isPending", ^{
-            context(@"when 'pending'", ^{
+        describe(@"isPendingActivation", ^{
+            context(@"when 'pendingActivation'", ^{
                 it(@"should return YES", ^{
-                    [[theValue([sut isPending]) should] beYes];
+                    NSLog(@"POOP: %@", [sut state]);
```
Comment (lausobo) [inline StateMachineTests/StateMachineSpec.m]: WAT

File: StateMachine/NSString+Capitalize.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//
+//  NSString+Capitalize.h
+//  StateMachine
+//
+//  Created by Steven Luscher on 2013-02-25.
+//  Copyright (c) 2013 Luis Solano Bonet. All rights reserved.
+//
```
Comment (lausobo) [inline StateMachine/NSString+Capitalize.h]: Could you please remove this headers?

File: StateMachine/NSString+Capitalize.m [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+@implementation NSString (Capitalize)
+- (NSString *)initialCapital
+{
+    NSString *firstLetter = [self substringToIndex:1];
+    return [self stringByReplacingCharactersInRange:NSMakeRange(0,1) withString:[firstLetter uppercaseString]];
+}
+- (NSString *)initialLowercase
+{
+    NSString *firstLetter = [self substringToIndex:1];
```
Comment (lausobo) [inline StateMachine/NSString+Capitalize.m:11]: Could you please add test coverage for this methods?

**Comment by lausobo** [comment]
@steveluscher thanks for the PR. Could you take a look at the comments please? Thanks!

File: StateMachine/NSString+Capitalize.m (hunk 3 above)
Comment (steveluscher) [inline StateMachine/NSString+Capitalize.m:11]: Done

**Comment by steveluscher** [comment]
Ready to merge, @luisobo?

**Pull request merged**